### PR TITLE
fix: {ATTENDEE} workflow variable always uses first attendee for for seated event

### LINKED ISSUE #10451: [CAL-2272] {ATTENDEE} workflow variable always uses first attendee for for seated event
### Issue Summary

When utilizing the "workflow" to send booking confirmation emails to attendees, the {ATTENDEE} variable consistently displays the name of the first attendee for all subsequent attendees. As a result, the 2nd and 3rd persons confirming the booking receive an email with the name of the first attendee instead of their own names.

<sub>[CAL-2272](https://linear.app/calcom/issue/CAL-2272/attendee-workflow-variable-always-uses-first-attendee-for-for-seated)</sub>

### PR DESCRIPTION
**Pull request by rkreddy99**
## What does this PR do?

Fixes an issue where {ATTENDEE} workflow variable always uses first attendee for for seated event.

Fixes #10451 

## Requirement/Documentation

- `scheduleEmailReminder` function doesnt use `sendTo` param for scheduling workflow reminders, instead always takes the first attendee from the event.
- so now check if sendTo attendee is present in evt.attendees, if yes use that otherwise fallback to evt.attendees[0].

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Schedule a workflow for seated event and check whether {ATTENDEE} variable is getting correctly replaced/

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by rkreddy99** [comment]
@CarinaWolli can you check whether this is the intended change, will make changes in the flow as per suggestions.

File: packages/features/ee/workflows/lib/reminders/emailReminderManager.ts [hunk 1]
```diff
@@ -74,9 +83,39 @@ export const scheduleEmailReminder = async (
...
+      } else if (typeof sendTo === 'object' && sendTo !== null) {
+        const { name, email } = sendTo;
+        attendeeEmailToBeUsedInMail = email
+      }
+    
+      for (const attendee of evt.attendees) {
+        if (attendeeEmailToBeUsedInMail === attendee.email) {
+          break
+        }
+      }
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/lib/reminders/emailReminderManager.ts:109]:
```suggestion
```
Can be removed

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
Works for emails, but we have the same issue SMS as well. Could you add that fix there too? 🙏

**Comment by rkreddy99** [comment]
@CarinaWolli made changes for sms

and some refactor in https://github.com/calcom/cal.com/pull/10468/commits/548f120f286c15a65000aeebec3164d2a92dbe5e, can you check this out.

**Review by CarinaWolli — COMMENTED** [review]
Thank you for addressing my feedback, I left you one more comment 🙏

File: packages/features/ee/workflows/lib/reminders/smsReminderManager.ts [hunk 2]
```diff
@@ -86,28 +88,49 @@ export const scheduleSMSReminder = async (
...
+    timeZone: string;
+    language: {
+      locale: string;
+    };
+  } | null = null;
+  if (action === WorkflowActions.SMS_ATTENDEE) {
+    const attendeeWithReminderPhoneAsSMSReminderNumber =
+      reminderPhone &&
+      evt.responses?.smsReminderNumber?.value === reminderPhone &&
+      evt.attendees.find((attendee) => attendee.email === evt.responses?.email?.value);
```
[the remark this answers is not in the document]
Comment (rkreddy99) [reply packages/features/ee/workflows/lib/reminders/smsReminderManager.ts:103]: Hey yup it should be enough, removing the redundant check

**Review by CarinaWolli — APPROVED** [review]
Great work, looks good now 🙌🏻 Thank you!

**Pull request merged**